packages/client/src/metrics_dependency_registry.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * The following provides functionality to enforce a consistent set of supported metrics dependencies and store them
+ * in a map for easy reference. This map can easily be extended to include additional state that may be required in
+ * the future like version, etc.
+ */
+
+/**
+ * maps dependency name like 'Amplitude' to corresponding DependencyInfo
+ */
+type DependencyMap = Map<string, DependencyInfo>;
+/**
+ * maps pkg name like 'amplitude-js' to corresponding DependencyInfo
+ */
+type PackageMap = Map<string, DependencyInfo>;
+
+export class DependencyInfo {
+  id: string;
+  pkg: string;
+  constructor(id: string, pkg: string) {
+    this.id = id;
+    this.pkg = pkg;
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-extraneous-class
+export abstract class MetricsDependencyRegistry {
+  static dependencyMap: DependencyMap = new Map<string, DependencyInfo>();
+
+  static packageMap: PackageMap = new Map<string, DependencyInfo>();
+
+  private static registerDependency(id: string, pkg: string): void {
+    this.dependencyMap.set(id, new DependencyInfo(id, pkg));
+    this.packageMap.set(pkg, new DependencyInfo(id, pkg));
+  }
+
+  static getDependencyMap(): DependencyMap {
+    return this.dependencyMap;
+  }
+
+  static getPackageMap(): PackageMap {
+    return this.packageMap;
+  }
+
+  static registerDependencies(): void {
+    const keys = Object.getOwnPropertyNames(this.prototype);
+    for (const key of keys) {
+      if (key === 'pkg' || key === 'id') {
+        const value = this.prototype[key];
+        if (typeof value === 'string') {
+          this.registerDependency(value, key);
+        }
+      }
+    }
+  }
+
+  protected static register(): void {}
+}
+
+interface IMetricsDependency extends MetricsDependencyRegistry {
```

Comment:
i didnt know that an interface can extend a class ?!

Reply:
so, you are doing this to maintain a registry ? why not simply call `MetricsDependencyRegistry.register` from every "register" call of a plugin ?